src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -9396,8 +9400,18 @@ void VRFTerminalUnitEquipment::ControlVRFToLoad(EnergyPlusData &state,
                 }
             } else {
                 getVRFTUZoneLoad(state, VRFTUNum, ZoneLoad, LoadToHeatingSP, LoadToCoolingSP, false);
-                if ((FullOutput < (LoadToHeatingSP - DataHVACGlobals::SmallLoad)) && !FirstHVACIteration) {
-                    SuppHeatCoilLoad = max(0.0, LoadToHeatingSP - FullOutput);
+                if (((FullOutput < (LoadToHeatingSP - DataHVACGlobals::SmallLoad) ||
+                      ((QZnReq - NoCompOutput) > DataHVACGlobals::SmallLoad && QZnReq <= 0.0))) &&
+                    !FirstHVACIteration) {
```

Comment:
Do you need this part? `((QZnReq - NoCompOutput) > DataHVACGlobals::SmallLoad && QZnReq <= 0.0)`
Won't `FullOutput` be equal to `NoCompOutput` here if the heating coil did not operate?
The blocks below are setting `SuppHeatCoilLoad` based on `FullOutput`, `NoCompOutput` isn't used below, so I don't think you need to check it here.

Reply:
The `FullOutput` may not be equal to `NoCompOutput` unless the cooling and heating coils are off. Sometimes the cooling coils needs to be `ON` for at least one of the zones. If you look defect files SPACE5-1, which is cooling dominated needs to be `ON`.

Reply:
But `FullOutput` and `NoCompOutput` are for the same zone if I'm following correctly. This is for a single terminal unit, not everyone connected to the same central unit.

Reply:
Yes, the `FullOutput` and `NoCompOutput` output variables are for the same terminal unit (or zone). These two variables may have the same value only in a specific situation. For instance, if the cooling coil is `OFF` and the parent is also operating in cooling mode, then these variables are equal. But there are instances where the values of these variables can be different and the logic needs to capture all those scenarios. Some of the cases may include:

-  for some zones the cooling coil is `OFF` and the parent is operating in cooling mode (these variables are equal) but the `NoCompOutput` may result in over-cooling. In this case the supplemental heater needs to kick in to temper the supply temperature to avoid over-cooling if needed.
- for other zones the cooling coil is `ON` and the parent is operating in cooling mode, (these variables are not equal). This scenario needs a different logic than the above.
- some zones may have the cooling and heating coils are `OFF` and the parent is operating in cooling mode (these variables are equal) but the zones may have heating load. In this the supplemental heater needs to meet the predicted heating load and the heating load due to outdoor air.
- and other scenarios

Reply:
I agree with this description. These checks are used to return if the TU does not need to turn on a coil. No/FullCompOutput does not tell the whole story. The QZnReq variable holds information on the zone load to meet. This `return` logic needs to account for all use cases (i.e., should be carefully implemented).